mirascope/core/base/prompt.py

```diff
@@ -307,18 +309,24 @@ async def run():
 _BasePromptT = TypeVar("_BasePromptT", bound=BasePrompt)
 _MessageParamT = TypeVar("_MessageParamT", bound=Any)
 _BaseDynamicConfigT = TypeVar("_BaseDynamicConfigT", bound=BaseDynamicConfig)
+_MessageFuncReturnTypes: TypeAlias = (
+    str | BaseMessageParam | list[BaseMessageParam] | list[Any]
+)
```

Comment:
If there are other return values that should be supported, add them here.

Reply:
It is possible that you and I have different ideas about the return type of a function.
I am assuming that list[Any] is the following case in the example you provided.
Obviously, `Image.Image` does not have a `BaseMessageParam`, so it allows Any.
In other words, the type you are defining here is the type that the function returns in `actual`, not the type after the decorator has made the change.

```python
@prompt_template
def recommend_book(previous_book: Image.Image) -> list[Any]:
    return ["I just read this book:", previous_book, "What should I read next?"]
```

The reason this type definition is necessary is so that the decorator can type-check all target functions that it is allowed to accept

Reply:
but the `prompt_template` decorator should require that the function it decorates returns the `Messages` type, which can be either `str`, or `Sequence[str, Image.Image]`, or `list[BaseMessageParam]`, or some form of `BaseDynamicConfig`.

the decorated function will then always have return type `list[BaseMessageParam]` or `BaseDynamicConfig` right?